src/marshmallow/fields.py

```diff
@@ -1243,6 +1243,9 @@ def _serialize(self, value: _D | None, attr, obj, **kwargs) -> str | float | Non
         return value.strftime(data_format)
 
     def _deserialize(self, value, attr, data, **kwargs) -> _D:
+        internal_type: type[_D] = getattr(dt, self.OBJ_TYPE)
+        if isinstance(value, internal_type):
+            return value
```

Comment:
This will let `datetime`s be accepted as `date`s because `datetime` is a `date` subclass (https://github.com/marshmallow-code/marshmallow/issues/1415#issuecomment-684948224).

This may not be an issue, since as a subclass it behaves like a `date`, but it may be surprising if the user then serialize the object and doesn't expect to see hours and minutes. At least this what I remember from when I encountered this in umongo.

Reply:
i think that is OK and well within an acceptable range of permissibility (along the lines of how `String` accepts `bytes`)